pkg/io/test/serializertest.go

```diff
@@ -49,10 +50,9 @@ func genericDecodeEncodeTest(t *testing.T, serializers ...perunio.Serializer) {
 		r.Close()
 		if err != nil {
 			t.Errorf("failed to decode %dth element (%T): %+v", i, v, err)
-		} else if !reflect.DeepEqual(v, dest.Interface()) {
-			t.Errorf(
-				"encoding and decoding the %dth element (%T) resulted in different value: %v, %v",
-				i, v, reflect.ValueOf(v).Elem(), dest.Elem())
+		} else {
+			_v := dest.Interface()
+			assert.Equalf(t, v, _v, "comparing element %d", i)
```

Comment:
Why did you change this?

Reply:
`assert.Equal` gives information on where the objects differ. This can be very helpful.

Reply:
I think it makes sense 👍🏻 We wrote this test very early on when we didn't use so much `assert`/`require`.

Note that it would now be nicer to `return` above after the `t.Error` and then move the `assert.Equalf` outside the `else` branch.

Reply:
This would change the functionality, right? Why is it nicer in your opinion?

Reply:
Ah I didn't see that this is happening inside a loop. nvm.